ggshield/output/text/iac_text_output_handler.py

```diff
@@ -0,0 +1,69 @@
+from io import StringIO
+from pathlib import Path
+from typing import ClassVar, List
+
+from ggshield.output.output_handler import OutputHandler
+from ggshield.scan import File, ScanCollection
+
+from ...core.text_utils import Line
+from ...core.utils import Filemode, get_lines_from_content
+from ...iac.models import IaCFileResult
+from .message import (
+    file_info,
+    iac_engine_version,
+    iac_vulnerability_header,
+    iac_vulnerability_location,
+)
+
+
+class IaCTextOutputHandler(OutputHandler):
+    nb_lines: ClassVar[int] = 3
+
+    def _process_scan_impl(self, scan: ScanCollection) -> str:
+        scan_buf = StringIO()
+        if scan.optional_header and (scan.iac_result or self.verbose):
+            scan_buf.write(scan.optional_header)
+
+        if scan.iac_result:
+            scan_buf.write(iac_engine_version(scan.iac_result.iac_engine_version))
+
+        if scan.iac_result:
+            for file_result in scan.iac_result.entities_with_incidents:
+                scan_buf.write(
+                    self.process_iac_file_result(
+                        Path(scan.id) / file_result.filename, file_result
+                    )
+                )
+
+        return scan_buf.getvalue()
+
+    def process_iac_file_result(
+        self, file_path: Path, file_result: IaCFileResult
+    ) -> str:
+        """
+        Build readable message on the found incidents.
+
```

Comment:
Can you add a description for  the `file_path` parameter as well ?

Reply:
Sure